### PR TITLE
Sequence load 8 (loader method tests and fixes)

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

Added tests for new TraceBaseLoader methods and fixed a couple issues they revealed.

Details:

- Added a check of existing column type definitions before checking that default values are of the correct type.
- Raised the exception about mssing required headers after having added the unknown headers error.

## Affected Issues/Pull Requests

- Partially addresses #824
- Merges into #874
- Next PR: #876

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/tests/utils/test_loader.py [hunk 1]
```diff
@@ -111,7 +114,7 @@ def __init__(self, *args, **kwargs):
     # handle_load_db_errors Tests
     def test_handle_load_db_errors_ve_choice(self):
         """Ensures handle_load_db_errors packages ValidationError about invalid choices"""
-        tl = self.TestLoader(None)
```
Comment (hepcat72) [inline DataRepo/tests/utils/test_loader.py:114]: C. The first arg is no longer required.

File: DataRepo/utils/loader.py [hunk 2]
```diff
@@ -1059,6 +1058,9 @@ def check_dataframe_headers(self, reading_defaults=False):
                         UnknownHeadersError(unknown_headers, file=file, sheet=sheet)
                     )
 
+        if len(missing_headers) > 0:
+            raise self.aggregated_errors_object
```
Comment (hepcat72) [inline DataRepo/utils/loader.py:1062]: C. I realized that I should raise all errors this method finds, not just the first one. It was useful to have the second.  It provided context.

**Review by lparsons — APPROVED** [review]
Fills in some test stubs. LGTM.

**Comment by hepcat72** [comment]
BTW, I accidentally had merged PR #874 into a stray branch I'd skipped (that had turned out to be identical), so I switched to the feature branch on my machine and merged in the branch from 874, then switched this PR to merge into the feature branch, so now that's fixed.

**Pull request merged**